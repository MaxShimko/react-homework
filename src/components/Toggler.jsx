import React from 'react';
import { Row, Col } from 'react-bootstrap';

import '../styles/Toggler.scss';

class Toggler extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            activeTab: 0
        };
    }

    toggleTab(id) {
        if (this.state.activeTab === id) return;

        this.setState({
           activeTab: id
        });
        this.props.onChange(id);
    }

    render() {
        return (
            <Row className='toggler'>
                <Col xs="auto" className="title">{this.props.title}</Col>
                <Col xs="auto" className="buttons">
                    {this.props.tabs.map((value, index) => {
                        return <div
                            key={index}
                            className={`${this.state.activeTab === index ? 'active' : ''}`}
                            onClick={_ => this.toggleTab(index)}>
                            {value}
                        </div>;
                    })}
                </Col>
            </Row>
        );
    }
}

export default Toggler;
