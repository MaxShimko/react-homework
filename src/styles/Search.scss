input {
  width: 100%;
  outline: none;
  border: none;
  border-radius: 4px;
  background: rgba(#232323, 0.6);
  color: #fff;
  padding: 10px 15px;
  box-shadow: 0 0 10px 0 rgba(#000, 0.5);
  backdrop-filter: blur(4px);
}

button {
  width: 100%;
  border: none;
  background: #F65261;
  color: #fff;
  border-radius: 4px;
  padding: 10px 15px;
  text-transform: uppercase;
  &:focus {
    outline: none;
  }
}
