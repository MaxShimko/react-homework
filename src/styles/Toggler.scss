.title {
  font-size: 14px;
  text-transform: uppercase;
  color: rgba(#fff, 0.8);
  padding-top: 5px;
  padding-bottom: 5px;
}

.buttons {
  display: flex;
  padding: 0;
  font-size: 14px;
  color: rgba(#fff, 0.8);
  user-select: none;

  div {
    padding: 5px 15px;
    background: rgba(#424242, 0.5);
    cursor: pointer;
    text-transform: uppercase;

    &:first-child {
      border-top-left-radius: 4px;
      border-bottom-left-radius: 4px;
    }
    &:last-child {
      border-top-right-radius: 4px;
      border-bottom-right-radius: 4px;
    }
  }

  .active {
    background: #F65261;
  }
}
