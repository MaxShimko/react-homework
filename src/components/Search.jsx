import React from 'react';
import { Row, Col } from 'react-bootstrap';

import '../styles/Search.scss';
import Toggler from './Toggler.jsx';

class Search extends React.Component {
    constructor(props) {
        super(props);

        this.filter = {type: 0, string: ''};

        this.state = {
            searchString: ''
        };
    }

    onChangeType(id) {
        this.filter.type = id;
        this.props.onSearch(this.filter);
    }

    onSearch() {
        this.filter.string = this.state.searchString;
        this.props.onSearch(this.filter);
    }

    handleInputChange(event) {
        this.setState({searchString: event.target.value});
    }

    render() {
        return <React.Fragment>
            <Row>
                <Col xs={8}>
                    <input type="text" placeholder="Search" value={this.state.searchString} onChange={this.handleInputChange.bind(this)} />
                </Col>
                <Col xs={4}>
                    <button onClick={this.onSearch.bind(this)}>Search</button>
                </Col>
            </Row>
            <div className="delimiter-2"></div>
            <Row>
                <Col>
                    <Toggler title="Search by" tabs={['Title', 'Genre']} onChange={this.onChangeType.bind(this)} />
                </Col>
            </Row>
        </React.Fragment>;
    }
}

export default Search;
