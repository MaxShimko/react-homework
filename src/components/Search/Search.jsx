import React from 'react';
import { Row, Col } from 'react-bootstrap';

import './style.scss';
import Toggler from '../Toggler/Toggler.jsx';

class Search extends React.Component {
    constructor(props) {
        super(props);

        this.filter = {type: 0, string: ''};

        this.state = {
            searchString: ''
        };
    }

    onChangeType(id) {
        this.filter.type = id;
        this.props.onSearch(this.filter);
    }

    onSearch() {
        this.filter.string = this.state.searchString;
        this.props.onSearch(this.filter);
    }

    handleInputChange(event) {
        this.setState({searchString: event.target.value});
    }

    render() {
        return (
            <React.Fragment>
                <Row>
                    <Col xs={8}>
                        <input id="test_search_input" type="text" placeholder="Search" value={this.state.searchString} onChange={this.handleInputChange.bind(this)} />
                    </Col>
                    <Col xs={4}>
                        <button id="test_search_button" onClick={this.onSearch.bind(this)}>Search</button>
                    </Col>
                </Row>
                <div className="delimiter-2" />
                <Row>
                    <Col>
                        <Toggler id="test_search_toggler" title="Search by" tabs={['Title', 'Genre']} onChange={this.onChangeType.bind(this)} />
                    </Col>
                </Row>
            </React.Fragment>
        );
    }
}

export default Search;
