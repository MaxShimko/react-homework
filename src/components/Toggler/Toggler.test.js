import React from 'react';
import { configure, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Toggler from './Toggler';
import {describe, expect, it} from '@jest/globals';
import Search from '../Search/Search';

configure({ adapter: new Adapter() });

describe('<Toggler />', () => {
    it('renders without crashing', () => {
        const component = render(<Toggler title="test title" tabs={['tab1', 'tab2']} />);
        expect(component).toMatchSnapshot();
    });

    it('right tab index', () => {
        const onChange = jest.fn()

        const component = mount(<Toggler title="test title" tabs={['tab1', 'tab2']} onChange={onChange} />);

        component.find('#test_toggler_tab_1').simulate('change', { target: {value: 'match string'} });
        component.find('#test_toggler_tab_1').simulate('click');
        expect(onChange).toHaveBeenCalledWith(1);
    });
});
