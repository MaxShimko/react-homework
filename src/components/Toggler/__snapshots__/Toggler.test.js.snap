// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggler /> renders without crashing 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "toggler row",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "title col-auto",
        },
        "children": Array [
          Object {
            "data": "test title",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "buttons col-auto",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "active",
                "id": "test_toggler_tab_0",
              },
              "children": Array [
                Object {
                  "data": "tab1",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "",
                  "id": "test_toggler_tab_1",
                },
                "children": Array [
                  Object {
                    "data": "tab2",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "",
                "id": "test_toggler_tab_1",
              },
              "children": Array [
                Object {
                  "data": "tab2",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "active",
                  "id": "test_toggler_tab_0",
                },
                "children": Array [
                  Object {
                    "data": "tab1",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "buttons col-auto",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "active",
              "id": "test_toggler_tab_0",
            },
            "children": Array [
              Object {
                "data": "tab1",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "",
                "id": "test_toggler_tab_1",
              },
              "children": Array [
                Object {
                  "data": "tab2",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "id": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "id": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "",
              "id": "test_toggler_tab_1",
            },
            "children": Array [
              Object {
                "data": "tab2",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "active",
                "id": "test_toggler_tab_0",
              },
              "children": Array [
                Object {
                  "data": "tab1",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "id": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "id": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "title col-auto",
          },
          "children": Array [
            Object {
              "data": "test title",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
